webapp/channels/src/components/widgets/modals/components/section_creator.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+import './section_creator.scss';
+
+type Props = {
+    title: {
+        id: string;
+        defaultMessage: string;
+    };
+    description?: {
+        id: string;
+        defaultMessage: string;
+    };
+    content: JSX.Element;
+    titleSuffix?: JSX.Element;
+};
+
+function SectionCreator({
+    title,
+    description,
+    content,
+    titleSuffix,
+}: Props): JSX.Element {
+    const {formatMessage} = useIntl();
+    const Title = (
+        <h4 className='mm-modal-generic-section__title'>
+            {formatMessage({id: title.id, defaultMessage: title.defaultMessage})}
+        </h4>
+    );
+
+    const Description = description && (
+        <p className='mm-modal-generic-section__description'>
+            {formatMessage({id: description.id, defaultMessage: description.defaultMessage})}
+        </p>
+    );
+
+    function titleRow() {
+        if (titleSuffix) {
+            return (<div className='mm-modal-generic-section__row'>
```

Comment:
0/5, but should we always have this row component here? It feels like it'll be easier to style consistently if that div is always there, regardless of whether or not there's a suffix item